// @vonage/applications/dist/applications.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Applications = void 0;
const server_client_1 = require("@vonage/server-client");
class Applications extends server_client_1.Client {
    authType = server_client_1.AuthenticationType.BASIC;
    async createApplication(application) {
        const resp = await this.sendPostRequest(`${this.config.apiHost}/v2/applications`, application);
        return resp.data;
    }
    async deleteApplication(applicationId) {
        await this.sendDeleteRequest(`${this.config.apiHost}/v2/applications/${applicationId}`);
    }
    async getApplication(applicationId) {
        const resp = await this.sendGetRequest(`${this.config.apiHost}/v2/applications/${applicationId}`);
        return resp.data;
    }
    async listApplications(filter) {
        const resp = await this.sendGetRequest(`${this.config.apiHost}/v2/applications`, filter);
        return resp.data;
    }
    async updateApplication(application) {
        const resp = await this.sendPutRequest(`${this.config.apiHost}/v2/applications/${application.id}`, application);
        return resp.data;
    }
}
exports.Applications = Applications;
